webapp/channels/src/components/header_footer_template_route.tsx

```diff
@@ -2,33 +2,37 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {Route} from 'react-router-dom';
+import {Route, RouteProps, RouteComponentProps} from 'react-router-dom';
 
 const HeaderFooterTemplate = React.lazy(() => import('components/header_footer_template'));
 const LoggedIn = React.lazy(() => import('components/logged_in'));
 
-export const HFTRoute = ({component: Component, ...rest}) => (
+interface Props extends RouteProps {
+    component: React.ComponentType<RouteComponentProps>;
+}
+
+export const HFTRoute = ({component: Component, ...rest}: Props) => (
     <Route
         {...rest}
         render={(props) => (
             <React.Suspense fallback={null}>
                 <HeaderFooterTemplate {...props}>
-                    <Component {...props}/>
+                    {Component && <Component {...props}/>}
```

Comment:
```suggestion
                    <Component {...props}/>
```
We do not need this check any more because component is never undefined as we it in above Props